engines/cryo/cryolib.h

```diff
@@ -0,0 +1,179 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef CRYO_CRYOLIB_H
+#define CRYO_CRYOLIB_H
+
+#include "audio/mixer.h"
+#include "common/system.h"
+
+#include "cryo/platdefs.h"
+
+namespace Cryo {
+
+class CryoEngine;
+
+#define SW16(n) ( (((n) & 0xFF) << 8) | (((n) >> 8) & 0xFF) )
+#define SW32(n) ( (((n) & 0xFF) << 24) | (((n) >> 24) & 0xFF) | (((n) & 0xFF00) << 8) | (((n) >> 8) & 0xFF00))
+#ifdef SCUMM_BIG_ENDIAN
+//big-endian host
+#define LE16(n) SW16(n)
+#define LE32(n) SW32(n)
+#define BE16(n) (n)
+#define BE32(n) (n)
+#else
+//little-endian host
+#define LE16(n) (n)
+#define LE32(n) (n)
+#define BE16(n) SW16(n)
+#define BE32(n) SW32(n)
+#endif
+
+enum {
+	fsFromStart = 1
+};
+
+struct BlitView{
+	int     _srcLeft;
+	int     _srcTop;
+	int     _dstLeft;
+	int     _dstTop;
+	int     _width;
+	int     _height;
+};
+
+class View {
+public:
+	View(int w, int h);
+	~View();
+
+	void setSrcZoomValues(int x, int y);
+	void setDisplayZoomValues(int w, int h);
+	void initDatas(int w, int h, void *buffer);
+	void centerIn(View *parent);
+
+	int      _width;
+	int      _height;
+	byte    *_bufferPtr;
+	int16    _pitch;
+	bool     _doubled;
+	BlitView _normal;
+	BlitView _zoom;
+};
+
+struct color3_t {
+	uint16   r, g, b;
```

Comment:
I don't get it... this game is using an indexed palette (256 colors), so these should be bytes, not 16-bit words. Same goes for the color_t struct below